magpie/api/management/user/user_utils.py

```diff
@@ -106,10 +112,42 @@ def _add_to_group(usr, grp):
         _add_to_group(new_user, _get_group(anonym_grp_name))
         new_user_groups.append(anonym_grp_name)
 
+    # Check for webhook requests
+    config_path = get_constant("MAGPIE_CONFIG_PATH", default_value=None,
+                               raise_missing=False, raise_not_set=False, print_missing=True)
+    if config_path:
+        webhook_configs = get_all_configs(config_path, "webhooks", allow_missing=True)
+        for cfg in webhook_configs:
+            if "create" in cfg.keys() and len(cfg["create"]) > 0:
+                # Execute all webhook requests
+                pool = multiprocessing.Pool(processes=len(cfg["create"]))
+                args = [(url, user_name) for url in cfg["create"]]
+                pool.starmap_async(webhook_request, args, error_callback=webhook_error_callback)
+
     return ax.valid_http(http_success=HTTPCreated, detail=s.Users_POST_CreatedResponseSchema.description,
                          content={"user": uf.format_user(new_user, new_user_groups)})
 
 
+def webhook_request(webhook_url, user_name):
+    # type: (Str, Str) -> None
+    """
+    Sends a webhook request using the input url.
+    """
+    # TODO: create a real temp_url that will be called if the webhook service has an error
+    #  this would also set the user's status to 0
```

Comment:
And what do we do if the callback is called after user deletion? I mean do we keep the user in the database just to signal that some webhooks failed uon deletion?

Reply:
I had the same question this morning :P
I talked with @fmigneault too about it. 
I wasn't sure what to do either, also considering the temp_url is not implemented yet.

The conclusion was that it was not the Magpie's responsability. If a failure occured, it would most probably be a user workspace folder that wouldn't be deleted, but the user would not be accessible anyway since he would be deleted from Magpie. At the same time, all the webhooks are not defined yet, and it could create other problems...

If we want it to be cleaner and safer, we would have to make the webhook requests blocking, and wait for their response to check if it was a success or not?

Reply:
I don't think neither that Magpie should prevent user deletion because of a third party. I was just wondering what you thought about it.
The consequence, however, is that I don't see the point in sending a callback url for the delete webhook.

Reply:
If we decide not to prevent the deletion, even with a failure of the webhooks, (as it is implemented right now), I shouldn't send the callback url, as the user won't exist anymore, and we won't be able to change the status.
I will remove the callback.